test/integration/agent/cniserver_test.go

```diff
@@ -777,6 +783,14 @@ func TestCNIServerChaining(t *testing.T) {
 		testRequire.Nil(err)
 		testRequire.Equal(tc.networkConfig, string(cniResp.CniResult))
 
+		// Check pod count metric
```

Comment:
I am definitely not in favor of piggybacking a metrics test in a highly "specialized" test like `TestCNIServerChaining`. There should be a dedicated test case for metrics.

Reply:
Thanks for the comment. I will try an alternative way.